### PR TITLE
initial guide in dashboard component

### PR DESCRIPTION
**Pull request by darrenmason**
screen shot 2015-07-16 at 6 35 32 pm

### CODE REVIEW

**Comment by darrenmason** [comment]
wip: weakest strongest, older, newer logic
wip: bars should be based on recent topics
wip: some responsive styling for sidebar width

File: src/components/student-dashboard/progress-guide.cjsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    colorClass = @colorizeBar(sectionPercent)
+    <div key={chapterSection} className='section'>
+      <div className='section-heading'>
+        <span className='section-number'>{chapterSection}</span>
+        <span className='section-title' title={section.title}>{section.title}</span>
+      </div>
+      <BS.ProgressBar className={colorClass} now={sectionPercent} />
+    </div>
+
+  colorizeBar: (percent) ->
```
Comment (philschatz) [inline src/components/student-dashboard/progress-guide.cjsx]: Should the progress bar rendering be in a common `.cjsx` file for both the dashboard widget and the Learning Guide?

Comment (nathanstitt) [inline src/components/student-dashboard/progress-guide.cjsx]: Yes, I think it should.  I'll refactor.

File: src/components/learning-guide.cjsx [hunk 2]
```diff
@@ -8,121 +8,24 @@ _ = require 'underscore'
...
-    sortWorst = _.sortBy(allSections, 'current_level')
-    sortWorst = sortWorst[0...4]
-
-    bestPages = _.flatten(_.pluck(sortBest, 'page_ids'))
-    worstPages = _.flatten(_.pluck(sortWorst, 'page_ids'))
-
-    weaker = _.map(sortWorst, @renderSectionBars)
-    stronger = _.map(sortBest, @renderSectionBars)
+    allSections = LearningGuideStore.getSortedSections(@props.courseId)
+    # if there are less than 4 sections, use 1/2 of the available ones
```
Comment (philschatz) [inline src/components/learning-guide.cjsx]: oh, nice!

**Pull request merged**